ui/src/TestAzureAccount.ts

```diff
@@ -0,0 +1,77 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { SubscriptionClient } from 'azure-arm-resource';
+import { SubscriptionListResult, TenantListResult } from 'azure-arm-resource/lib/subscription/models';
+import { ApplicationTokenCredentials, AzureEnvironment, loginWithServicePrincipalSecret } from 'ms-rest-azure';
+import { Event, EventEmitter } from 'vscode';
+import { AzureAccount, AzureLoginStatus, AzureResourceFilter, AzureSession, AzureSubscription } from './azure-account.api';
+import { ArgumentError } from './errors';
+import { localize } from './localize';
+
+export class TestAzureAccount implements AzureAccount {
```

Comment:
As mentioned in the Functions PR, I think this class should provide a way to get subscription information (at least subscription id and credentials) so that people can validate Azure stuff without going through our tree